model/models.py

```diff
@@ -122,9 +121,68 @@ def forward(self, sub, rel):
 		x				= self.hidden_drop2(x)
 		x				= self.bn2(x)
 		x				= F.relu(x)
-
 		x = torch.mm(x, all_ent.transpose(1,0))
 		x += self.bias.expand_as(x)
 
 		score = torch.sigmoid(x)
 		return score
+
+class CompGCN_ConvKD(CompGCNBase):
+	def __init__(self, edge_index, edge_type, params=None):
+		super(self.__class__, self).__init__(edge_index, edge_type, params.num_rel, params)
+
+		#self.bn0 = torch.nn.BatchNorm2d(1)
+		self.drop = torch.nn.Dropout(self.p.hid_drop)
+
+		self.relu = torch.nn.ReLU()
+		self.conv2d1 = torch.nn.Conv2d(1, out_channels=self.p.num_filt, kernel_size=(1, 3), stride=1)
+		self.fake_fc_with_conv = torch.nn.Conv2d(self.p.num_filt,out_channels= 1 , kernel_size=(self.p.embed_dim, 1), stride=self.p.embed_dim,bias=False)
+
+		total_params = sum(
+			param.numel() for param in self.parameters()
+		)
+		print("Total Parameter: " + str(total_params))
+
+	def forward(self, sub,rel):
+		sub_emb, rel_emb, all_ent = self.forward_base(sub, rel, self.drop, self.drop)
+		h = sub_emb
+		r = rel_emb
+		t = all_ent
+		t = self.drop(t)
+		# bs x 1 x dim
+		l = len(sub)
+
+
+
+		h = h.unsqueeze(1).expand(-1,self.p.num_ent,-1).reshape((l,1,self.p.num_ent * self.p.embed_dim))
+		r = r.unsqueeze(1).expand(-1,self.p.num_ent,-1).reshape((l,1,self.p.num_ent * self.p.embed_dim))
+		t = t.unsqueeze(0)
+		t = t.expand(l,-1,-1).reshape((l,1,self.p.num_ent * self.p.embed_dim))
```

Comment:
Please use identifiers, which allow to quickly identify what this is, instead of single letter names